File: Design-Patterns/Strategy/CSharp/AttributeStrategies/MagicaAttributeStrategy.cs
namespace Strategy.AttributeStrategies
{
    public class MagicaAttributeStrategy : AttributeStrategy
    {
        public override int Health => Properites.Level * Properites.Strength;
        public override int Mana => Properites.Level * Properites.Wisdom * 5;
        public override int Attack => Properites.Level + Properites.Strength;
        public override int MagicalAttack => Properites.Level + Properites.Wisdom * 10;
        public override int Defense => Properites.Level + Properites.Strength;
        public override int MagicalDefense => Properites.Level + Properites.Wisdom * 5;
    }
}

File: Design-Patterns/Strategy/Cpp/magica_attribute_strategy.cpp
#include "attribute_strategy.h"

int magica_attribute_strategy::get_health()
{
	return properties_->level * properties_->strength;
}

int magica_attribute_strategy::get_mana()
{
	return properties_->level * properties_->wisdom * 5;
}

int magica_attribute_strategy::get_attack()
{
	return properties_->level + properties_->strength;
}

int magica_attribute_strategy::get_magical_attack()
{
	return properties_->level + properties_->wisdom * 10;
}

int magica_attribute_strategy::get_defense()
{
	return properties_->level + properties_->strength;
}

int magica_attribute_strategy::get_magical_defense()
{
	return properties_->level + properties_->wisdom * 5;
}
